specification/common-types/resource-management/v1/types.json

```diff
@@ -319,6 +319,110 @@
         }
       },
       "description": "The resource management error additional info."
+    },
+    "JSONWebKey": {
```

Comment:
Why is this going in common-types? It does not appear to be universal to all RPs?

Reply:
I will make the JSONWebKey definition in our own RPs.

Reply:
Where do protocol elements defined in an IETF RFC go? I would think that elements defined in an RFC are by definition "common" and are not protocol specific. If there's a better place under common-types than types.json for this definition, please let us know.

Reply:
Oh, and Amy: Please don't make the changes to turn these definitions private until we get clear guidance that IETF defined types are not "common" types (or guidance about where we should put internet standard type definitions). It seems rather strange that the SDK requires private definitions of these standardized types.

Reply:
@LarryOsterman  "common" in this case is a reference to cross-RP usage patterns. Are there other RPs that will be using this type in the foreseeable future?

Reply:
I don't know the answer to your question. In general, standards are intended to be used by multiple customers, so everyone who implements RFC7517 or RFC7515 (which depends on 7515) is likely to require this. These are widely used internet protocols, so it seems likely that there are multiple teams that would require this. I do know that AAD and AKV for certain use this protocol and I'm willing to bet there are many more azure services that are going to use them. However i don't know if those services are resource providers.

If I was designing the layout for files in the azure SDK, I would have an internet standards folder indexed by RFC number which would host the swagger documents for internet protocols. That way any RPs which want to implement JSON based internet protocols would have a place to put their swagger documents.

Reply:
PS: I also suspect that it is highly likely that one or more azure services implement internet standard protocols So having support in the azure SDK for internet protocols seems like a good thing.

Reply:
I'm in favor of making this reusable, but agree with @KrisBash's point about the specific file it's going into - `.../resource-management/v1/types.json` are  generally types owned by ARM relating to resources.

What about creating a new file for it in a different location - `specification/common-types/rfcs/rfc7517.json` (or something more user friendly)?

@MyronFanQiu what are your thoughts?

Reply:
@LarryOsterman @MyronFanQiu  I am ok with the proposal Larry and Anthony suggested of having a common definition directory for rfcs. I think it's not strictly necessary at this point as I am not aware of other RPs using this type, but it's not a bad precedent. I am opposed to modifying types.json. Ownership of this file needs to be retained by ARM as changes to this file can be very disruptive to many providers.

Reply:
I'm totally happy with anthony's suggestion.

And there are already RPs using JSON Web Keys in their protocols.

For instance I found
specification\containerregistry\data-plane\Microsoft.ContainerRegistry\preview\2018-08-10\containerregistry.json 
which has a private definition for JWKs. And AKV uses x5c in its protocol elements.

My suspicion is that they ran into the same issue as flagged here and they weren't as obnoxious about API private definitions.

Reply:
JSONWebKeySet definition has been moved to a specific folder: specification/common-types/rfcs/rfc7517.json